voice/voice-ai/x/SpeechRecognition/SpeechRecognition.swift

```diff
@@ -750,6 +770,8 @@ class SpeechRecognition: NSObject, ObservableObject, SpeechRecognitionProtocol {
                 self.recognitionLock.signal()
                 
                 // Make the query on the background thread
+                self.timeLogger?.finishSTTEnd()
```

Comment:
I think we are doing the end-of-stt part slightly incorrectly. But we are getting away with it because users tend to wait a bit  after they finish speaking, before they release the press and hold button. We should have waited for the full STT to finish, then copy the text. Instead what we do now is immediately take whatever text is available and make query based on that, because `self.recognitionTask?.finish()` most likely is non-blocking.  That means we need to change the handler in `handleRecognition` let that function signals when the final STT is ended. We can still keep `timeLogger.finishSTTEnd()` here. The issue can be fixed later.